### ISSUE TITLE
Switch statements

### ISSUE THREAD

**Issue by KCreate** [KCreate/charly-lang#118]
```javascript
let myproperty = 25
switch myproperty {
  case 10 {}
  case 20 {}
  case 25, 30, 40 {}
  default {}
}
```

default case is optional. Multiple values can be matched using a comma.

The above is equivalent to:

```javascript
let myproperty = 25
let test = myproperty == 10 // this is being generated internally
if test == 10 {

} else if test == 20 {

} else if test == 25 || test == 30 || test == 40 {

} else {
  // default block
}
```

**Comment by KCreate** [comment]
The test case should first be put in a temporary variable to avoid duplicate evaluation of call expressions.

**Comment by KCreate** [comment]
Implemented in #157
